eth_utils/logging.py

```diff
@@ -73,13 +73,18 @@ def get_logger(name: str, logger_class: Type[TLogger] = None) -> TLogger:
             # accidentally return the incorrect logger type because the logging
             # module does not *update* the cached instance in the event that
             # the global logging class changes.
-            if name in logging.Logger.manager.loggerDict:
-                if type(logging.Logger.manager.loggerDict[name]) is not logger_class:
-                    del logging.Logger.manager.loggerDict[name]
+            #
+            # types ignored b/c mypy doesn't identify presence of manager on logging.Logger
+            if name in logging.Logger.manager.loggerDict:  # type: ignore
+                if (
+                    type(logging.Logger.manager.loggerDict[name])  # type: ignore
+                    is not logger_class
+                ):
+                    del logging.Logger.manager.loggerDict[name]  # type: ignore
             return logging.getLogger(name)
 
 
-def get_extended_debug_logger(name: str) -> ExtendedDebugLogger:
+def get_extended_debug_logger(name: str) -> logging.Logger:
```

Comment:
This feels wrong to me. Since this returns a `ExtendedDebugLogger` and `ExtendedDebugLogger` derives from `logging.Logger` the return type should be `ExtendedDebugLogger`. I'm 99 % certain this will cause problems at every place that calls `logger = get_extended_debug_logger('foo')` and then tries to call `logger.debug2("meh"). It will rightfully yell that `debug2` is not a method on `logging.Logger` because we are hiding the fact that what we return is in fact an `ExtendedDebugLogger`

Reply:
Yeah, the problem I'm having is with `get_logger() -> logging.Logger`. With...
> def get_extended_debug_logger(name: str) -> ExtendedDebugLogger:

> def get_logger(name: str, logger_class: Type[logging.Logger] = None) -> TLogger:
> errors....
> eth_utils/logging.py:68: error: Incompatible return value type (got "Logger", expected "TLogger")
> eth_utils/logging.py:84: error: Incompatible return value type (got "Logger", expected "TLogger")

or with 

>def get_logger(name: str, logger_class: Type[logging.Logger] = None) -> logging.Logger:
> eth_utils/logging.py:88: error: Incompatible return value type (got "Logger", expected "ExtendedDebugLogger")

Reply:
I ended up just casting the return value in `get_extended_debug_logger()` to `ExtendedDebugLogger` -  so everything works, but there might be a better solution.

Reply:
Yeah, fine by me. The bound looks correct. Probably worth following up on this because it suggests `get_logger` does not preserve the type as we want it to. Anyway, the cast is an acceptable solution for now imho.